src/app/qgsmaptooladdfeature.cpp

```diff
@@ -121,31 +121,45 @@ void QgsMapToolAddFeature::featureDigitized( const QgsFeature &feature )
     }
     if ( topologicalEditing )
     {
-      const QList<QgsPointLocator::Match> sm = snappingMatches();
-      for ( int i = 0; i < sm.size() ; ++i )
+      const QVector<QgsVectorLayer *> layers = vlayer->project()->layers<QgsVectorLayer *>();
+
+      for ( QgsVectorLayer *layer : layers )
       {
-        if ( sm.at( i ).layer() && sm.at( i ).layer()->isEditable() && sm.at( i ).layer() != vlayer )
+        if ( !layer->isEditable() || layer == vlayer )
```

Comment:
Do we still need to differentiate between current layer and others? We're applying the same logic now, except for the begin/endEditCommand.